mdal/frmts/mdal_hec2d.cpp

```diff
@@ -761,13 +766,11 @@ std::unique_ptr<MDAL::Mesh> MDAL::DriverHec2D::load( const std::string &resultsF
   {
     HdfFile hdfFile = openHdfFile( mFileName );
 
-    // Verify it is correct file
-    std::string fileType = openHdfAttribute( hdfFile, "File Type" );
-    bool oldFormat = canReadOldFormat( fileType );
-
     HdfGroup gGeom = openHdfGroup( hdfFile, "Geometry" );
     HdfGroup gGeom2DFlowAreas = openHdfGroup( gGeom, "2D Flow Areas" );
 
+    bool oldFormat( existsHdfDataset( gGeom2DFlowAreas, "Names" ) );
```

Comment:
are you sure that this format you want to support is exactly same as "old format" in all other spec? if it is different format that 2 we currently support ("old" and "new"), maybe there is need to add 3rd format spec .... ?